### ISSUE TITLE
Does not work with the new Visual Studio editor

### ISSUE THREAD

**Issue by mjrichards91** [DarkCloud14/MonoDevelop.StyleCop#27]
Latest updates to Visual Studio (8.1.5 (build 9) for instance) automatically chooses to use the new editor and it does not work with StyleCop. The menu option in the right-click menu no longer appears and running it from Tools -> Run StyleCop shows "Violations: 0" where there should be some.

Reverting to the old editor allows it to work again.

![image](https://user-images.githubusercontent.com/3430893/61311778-0c674080-a7c5-11e9-82f7-60fe9f74ceeb.png)

**Comment by DarkCloud14** [comment]
I'll have a look at it but I can't garuantee to get all functionality back the old editor has like showing the warnings properly at the line they occur. Running it from Tools or right click menu of the current files tabulator works fine for me so far, you use the latest version of the addin (1.1.0.2), if not upgrade please.

Maybe its best to switch to the CodeAnalyzers which make use of roslyn as VS for Mac goes more and more into that direction and I won't make that switch for this StyleCop addin. If you don't know the roslyn based StyleCop Analyzers yet, have a look at it [here](https://github.com/DotNetAnalyzers/StyleCopAnalyzers).

You'll have to create the configuration manually as far as I know as I think it isn't compatible with the old StyleCop configuration file but I guess sooner or later (sooner) I'll give up on this project as it would need a rewrite that I can't do anymore, also the currently used StyleCop version isn't the latest anymore I guess but as that project is also almost dead in favor of the StyleCop.Analyzers project and the roslyn based approach there isn't such a big gap between the version in use here and the latest released StyleCop version.

**Comment by DarkCloud14** [comment]
FYI if you have a standard .Net project and add StyleCop.Analyzers NuGet package you must add the following ItemGroup to your project file manually (at least when using VS for Mac) as for some reason it won't do that automatically:
`  <ItemGroup>
    <Analyzer Include="packages\StyleCop.Analyzers.1.1.118\analyzers\dotnet\cs\StyleCop.Analyzers.CodeFixes.dll" />
    <Analyzer Include="packages\StyleCop.Analyzers.1.1.118\analyzers\dotnet\cs\StyleCop.Analyzers.dll" />
  </ItemGroup>`

Depending on where the project file is on file system you must also correct the path to the dlls..
After that you can build the project and the analyzer will do its job. The good thing using this analyzer is that it'll also provide code fixes.

**Comment by DarkCloud14** [comment]
Hello again, I tested a bit more and tried to get the StyleCop warnings/errors somehow shown in the new C# editor but wasn't succesful. The old editor does the display of the warnings/errors itself without that I've to do anything else than adding the warnings/errors to error pad, but the new doesn't seem to do that and I don't know how to influence that.

I also get the warnings/errors of StyleCop in error pad with the new editor, only displaying it in the editor doesn't work in the new editor anymore, but that is in my opinion a problem of the new editor. As said above I didn't have to do anything but add the warnings/errors to error pad to get the old editor show these.

I'll close this bug as at least the missing context menu entries are added back with the latest version 1.1.0.3 which will also add support for VS for Mac 8.2 (or better it only works with that version).

If you need more please consider to switch to the StyleCop.Analyzers NuGet package (as mentioned above) which is still under development...

**Comment by mjrichards91** [comment]
I think this will be fine for the time being. Thank you for your work on this!

**Comment by DarkCloud14** [comment]
FYI I tested StyleCop.Analyzers NuGet package again and I think I was wrong, while you have a file open and are editing, it'll show you problems and suggest quick fixes in editor window but not in error pad.

To see problems in error pad you must build the project.

Unfortunately you might still have to do the manual editing of the project file mentioned above.
I tested with a new .Net 4.7 project and there it works without manual editing but adding the NuGet package to the MonoDevelop.StyleCop project requires me to add the above mentioned lines manually for whatever reason but then the analyzer works.
